tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/exceptions.scala

```diff
@@ -33,7 +33,8 @@ class TypeException(message: String, causeExpr: TlaEx) extends CheckerException(
   * @param errors the list of type inference errors
   */
 class TypeInferenceException(val errors: Seq[TypeInferenceError])
-  extends CheckerException("Type inference failed",
+  extends CheckerException("Type inference failed, first error: %s"
+      .format(if (errors.nonEmpty) errors.head.explanation else "unknown"),
     if (errors.nonEmpty) errors.head.origin else NullEx)
```

Comment:
Good point. But the second conditional constructs the second argument of the exception constructor. So it will stay like this. Though I agree it looks a bit untidy.